### PR TITLE
docs: remove redundant env docs

### PR DESCRIPTION
**Pull request by camargo**
## Description

* For now the main one will remain in the deployment directory
* See: https://github.com/NASA-AMMOS/aerie/blob/develop/deployment/Environment.md

### CODE REVIEW

File: docs/source/contribute/quickstart/index.rst [hunk 1]
```diff
@@ -16,7 +16,6 @@ Prerequisite Software
   :includehidden:
 
   build
-  environment
```
Comment (Mythicaeda) [inline docs/source/contribute/quickstart/index.rst:19]: Can you update this to [a link to the external file instead](https://github.com/NASA-AMMOS/aerie/blob/develop/deployment/Environment.md)? An example of how to do so can be found in `docs/source/contribute/index.rst`, specifically on the last line of the file.

File: docs/source/user-guide/quickstart/index.rst [hunk 2]
```diff
@@ -10,7 +10,6 @@ Prerequisite Software
   :includehidden:
 
   install
-  environment
```
Comment (Mythicaeda) [inline docs/source/user-guide/quickstart/index.rst:13]: Same request here as in `/contribute/quickstart/index.rst`

**Review by Mythicaeda — COMMENTED** [review]
Gah, didn't mean to leave a top-level comment, sorry

File: docs/source/contribute/quickstart/index.rst (hunk 1 above)
Comment (camargo) [reply docs/source/contribute/quickstart/index.rst:19]: Done

File: docs/source/user-guide/quickstart/index.rst (hunk 2 above)
Comment (camargo) [reply docs/source/user-guide/quickstart/index.rst:13]: Done

**Review by Mythicaeda — APPROVED** [review]

**Pull request merged**